aws/userdata/make-binary-package.sh

```diff
@@ -53,27 +53,26 @@ export VERSION
 export IS_NIGHTLY
 
 aws s3 sync "s3://hhvm-nodist/${DISTRO}/" nodist/
-if ! bin/make-package-in-throwaway-container "$DISTRO"; then
-  # On modern systems, this should just be:
-  #   $(aws ecr get-login --no-include-email --region us-west-2)
-  # This is slightly different to support the older versions of the AWS and
-  # docker CLIs in our base image (currently Ubuntu 16.04)
-  $(aws ecr get-login --region us-west-2 | sed 's/ -e none / /')
-
-  FAILED_CONTAINER_ID="$(docker ps -aq)"
-  EC2_INSTANCE_ID="$(curl --silent http://169.254.169.254/latest/meta-data/instance-id)"
-
-  # Create a Docker image from the container (instance)
-  DOCKER_REPOSITORY="223121549624.dkr.ecr.us-west-2.amazonaws.com"
-	IMAGE_NAME="${DOCKER_REPOSITORY}/hhvm-failed-builds:${VERSION}_${DISTRO}_${EC2_INSTANCE_ID}"
-  docker commit "${FAILED_CONTAINER_ID}" "${IMAGE_NAME}"
-  # Push to ECR so we can download to debug
-	docker push "${IMAGE_NAME}"
-
-	# Once we're confident with this, shutdown immediately instead of waiting for
-  # the timeout
-	exit 1
+if bin/make-package-in-throwaway-container "$DISTRO"; then
+  IMAGE_NAME=hhvm-succeeded-builds
+else
+  IMAGE_NAME=hhvm-failed-builds
 fi
+# On modern systems, this should just be:
+#   $(aws ecr get-login --no-include-email --region us-west-2)
+# This is slightly different to support the older versions of the AWS and
+# docker CLIs in our base image (currently Ubuntu 16.04)
+$(aws ecr get-login --region us-west-2 | sed 's/ -e none / /')
+
+CONTAINER_ID="$(docker ps -aq)"
+EC2_INSTANCE_ID="$(curl --silent http://169.254.169.254/latest/meta-data/instance-id)"
```

Comment:
this might be available somewhere under /var/lib/cloud/instance but I'm not sure

Reply:
This endpoint appears to be the preferred way, based on https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html